### PR TITLE
Napatech bug fix 01

### PR DESCRIPTION
**Pull request by pjyoung1**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- This fixes an issue where flows programmed to the card sometimes failed to be entered into the flow table with no indication of an error.  The root cause was a handle that was shared between different threads.  Having each thread get it's own handle fixes the issue.

- Fix an issue with SPAN port configuration where comparing the byte order of the src and dest address in an IPv4 address was incorrect due to endian issues.  wrapping these addresses in htonl() fixes the issue.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: src/source-napatech.c [hunk 1]
```diff
@@ -359,6 +359,35 @@ static int CompareIPv6Addr(uint8_t addr_a[16], uint8_t addr_b[16]) {
...
+ * flows to bypass.
+ *
+ * \return the flow stream handle, NULL if failure.
+ */
+static NtFlowStream_t InitFlowStream(int adapter, int stream_id)
+{
+    int status;
+    NtFlowStream_t hFlowStream;
+
+        NtFlowAttr_t attr;
```
Comment (jlucovsky) [inline src/source-napatech.c:376]: nit: indentation

File: src/source-napatech.c [hunk 2]
```diff
@@ -770,17 +789,29 @@ TmEcode NapatechPacketLoop(ThreadVars *tv, void *data, void *slot)
     uint64_t pkt_ts;
     NtNetBuf_t packet_buffer;
     NapatechThreadVars *ntv = (NapatechThreadVars *) data;
+    NtFlowStream_t flow_stream[MAX_ADAPTERS];
```
Comment (jlucovsky) [inline src/source-napatech.c:792]: Suggest moving this within the guard `#ifdef NAPATECH_ENABLE_BYPASS`.
Suggest using `= {0}` to zero-fill.

File: src/source-napatech.c [hunk 3]
```diff
@@ -987,9 +1021,6 @@ TmEcode NapatechPacketLoop(ThreadVars *tv, void *data, void *slot)
     } // while
 
     if (closer) {
-#ifdef NAPATECH_ENABLE_BYPASS
-        NapatechCloseFlowStreams();
```
Comment (jlucovsky) [inline src/source-napatech.c:991]: Remove function prototype from `util-napatech.h`.

File: src/util-napatech.c [hunk 4]
```diff
@@ -46,7 +46,7 @@ typedef struct FlowStatsCounters_
     uint16_t total_bypass_flows;
 } FlowStatsCounters;
 
-static NtFlowStream_t hFlowStream[MAX_ADAPTERS];
+//static NtFlowStream_t hFlowStream[MAX_ADAPTERS];
```
Comment (jlucovsky) [inline src/util-napatech.c:49]: Delete if not needed.

File: src/util-napatech.c [hunk 5]
```diff
@@ -83,21 +83,21 @@ static int GetNumAdapters(void)
...
+int NapatechVerifyBypassSupport(void)
 {
     int status;
     int adapter = 0;
-    int num_adapters = GetNumAdapters();
+    int num_adapters = NapatechGetNumAdapters();
     SCLogInfo("Found %d Napatech adapters.\n", num_adapters);
-    memset(&hFlowStream, 0, sizeof(hFlowStream));
+    NtFlowStream_t hFlowStream;
+    //    memset(&hFlowStream, 0, sizeof(hFlowStream));
```
Comment (jlucovsky) [inline src/util-napatech.c:100]: Delete if not needed.

File: src/util-napatech.c [hunk 6]
```diff
@@ -113,50 +113,18 @@ int NapatechInitFlowStreams(void)
...
     bypass_supported = 1;
     return bypass_supported;
 }
 
-/**
- * \brief  Returns a pointer to the FlowStream associated with this adapter.
- *
- * \return count of the Napatech adapters present in the system.
- */
-NtFlowStream_t *NapatechGetFlowStreamPtr(int device)
```
Comment (jlucovsky) [inline src/util-napatech.c:132]: Remove prototype from `util-napatech.h`

File: src/util-napatech.c [hunk 7]
```diff
@@ -113,50 +113,18 @@ int NapatechInitFlowStreams(void)
...
-{
-    return &hFlowStream[device];
-}
-
-/**
- * \brief Closes all open FlowStreams
- *
- * \return Success of the operation.
- */
-int NapatechCloseFlowStreams(void)
```
Comment (jlucovsky) [inline src/util-napatech.c:142]: ditto

File: src/util-napatech.c [hunk 8]
```diff
@@ -59,7 +59,7 @@ int NapatechIsBypassSupported(void)
  *
  * \return count of the Napatech adapters present in the system.
  */
-static int GetNumAdapters(void)
+int NapatechGetNumAdapters(void)
```
Comment (jlucovsky) [inline src/util-napatech.c:62]: This is called several times but the value is invariant. Suggest fetching value once and returning that value on subsequent calls.

File: src/util-napatech.c [hunk 9]
```diff
@@ -453,7 +421,7 @@ static uint32_t UpdateStreamStats(ThreadVars *tv,
     uint64_t total_dispatch_fwd_pkts = 0;
     uint64_t total_dispatch_fwd_byte = 0;
 
-    for (adapter = 0; adapter < GetNumAdapters();  ++adapter) {
+    for (adapter = 0; adapter < NapatechGetNumAdapters();  ++adapter) {
```
Comment (jlucovsky) [inline src/util-napatech.c:424]: nit: indentation off in this function.

**Comment by victorjulien** [comment]
Replaced by #4994

**Pull request closed without merging**